### DISCUSSION TITLE
I am trying to create a container and embed child's using react and typescript

### DISCUSSION THREAD

**Discussion by dkchaitanya92** [clientIO/joint#2133] [Q&A]
### What happened?

I am trying to create a container and embed childs using react typescript. Taking reference of your https://www.jointjs.com/demos/container 

Getting error while creating child using joint.shapes.container.Base.define('container.Child',{}).

codesandbox reference code:

https://codesandbox.io/s/reactts-container-lqsb14?file=/src/Diagram.tsx

### Version

joint js+ v3.4

### What browsers are you seeing the problem on?

Firefox, Chrome

### What operating system are you seeing the problem on?

Windows, Linux

**Comment by kumilingus** [comment]
You didn't define `container.Base` (it's defined [here](https://github.com/clientIO/joint/blob/master/demo/container/src/joint.shapes.container.js#L19) just before `container.Child`).

**Comment by kumilingus** [comment]
Please use our discussion [forum](https://github.com/clientIO/joint/discussions) for similar technical questions.

**Comment by MeAriJit** [comment]
Hello @kumilingus, I am facing the same issue with react. Could you please again take a look.

 I can see he did define container.Base just before [container.child.] on line number 37 in  https://codesandbox.io/s/reactts-container-lqsb14?file=/src/Diagram.tsx

Somehow you missed the code. please take a look.

**Comment by kumilingus** [comment]
Hi @MeAriJit , my bad. Here's updated [example](https://codesandbox.io/s/reactts-container-forked-hbcwhm?file=/src/Diagram.tsx).

**Comment by MeAriJit** [comment]
Hello @kumilingus 
Thanks for your update.
Now I am able to add contaier.Base. But facing some issue implementing your container example code.
Issue:
1. Getting error at line no. 262 this.fitToCHildren of shapes.containers.Parent({}) error:  // error : this.fitToCHildren not a function 


Now I Bypassed the error through commenting the error generating code mentioned at Issues 1.
But still facing following issues :
1. Output is not as expected but Code base is same as https://www.jointjs.com/demos/container your demo 
https://codesandbox.io/s/reactts-container-hcd37q?file=/src/Diagram.tsx

3. While trying to collapse the Base : error occuring as below
this.fitParent is not a function
child.fitAncestorElements
/src/Diagram.tsx:64:17
  61 | {
  62 |   fitAncestorElements: function () {
  63 |     var padding = 10;
> 64 |     this.fitParent({
     |         ^
  65 |       deep: true,
  66 |       padding: {
  67 |         top: headerHeight + padding,
Is it somehow we need to bind this? 
Please have a look.

**with commented out error code**
codesandbox:
https://codesandbox.io/s/reactts-container-hcd37q?file=/src/Diagram.tsx

**with error at  this.fitToCHildren or Parent** 
codesandbox:
https://codesandbox.io/s/reactts-container-lqsb14?file=/src/Diagram.tsx

we are in process of getting the license and this container and collapse  feature is essential for us to use in jointjs+ with react.
Please help on this poc.

**Comment by kumilingus** [comment]
`fitParent` and `fitToChildren` are not published yet. They will be available in v3.7.

You should use the code from the latest release `v3.6.5` [here](https://github.com/clientIO/joint/tree/v3.6.5/demo/container).
